skimage/measure/_regionprops.py

```diff
@@ -413,10 +413,18 @@ def __getattr__(self, attr):
                     f"Attribute '{attr}' unavailable when `intensity_image` "
                     f"has not been specified."
                 )
+            warn(
+                f"`RegionProperties.{attr}` is deprecated starting in "
+                "version 0.26 and will be removed in version 2.0. Use "
+                f"`RegionProperties.{PROPS[attr]}` instead. ",
+                category=FutureWarning,
+                stacklevel=2,
+            )
             # retrieve deprecated property (excluding old CamelCase ones)
             return getattr(self, PROPS[attr])
-        else:
-            raise AttributeError(f"'{type(self)}' object has no attribute '{attr}'")
+
+        # Fallback to default behavior, potentially raising an attribute error
+        return self.__getattribute__(attr)
```

Comment:
`{type(self)}` would print `<class 'skimage.measure._regionprops.RegionProperties'>` here which isn't really what we want.

---

Aside: I noticed that something magically appends

> Did you mean '...'

to the previous error message. Apparently only for properties. E.g.

```python
class A:
    @property
    def a(self):
        return 3
    
    def __getattr__(self, item):
        raise AttributeError("not found")
    
A().A
```
```
[...] line 7, in __getattr__
    raise AttributeError("not found")
AttributeError: not found. Did you mean: 'a'
```

If that is official Python behavior I can't find a reference to this behavior? Am I crazy? 🤯

Reply:
What version of Python are you seeing this on? I don't see it with 3.13.

Reply:
Weird, I see it on 3.13.3 inside a simple repl:
```python
Python 3.13.3 (main, Apr  9 2025, 07:44:25) [GCC 14.2.1 20250207] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class A:
...     @property
...     def a(self):
...         return 3
...     
...     def __getattr__(self, item):
...         raise AttributeError("not found")
...     
... A().A
... 
Traceback (most recent call last):
  File "<python-input-0>", line 9, in <module>
    A().A
  File "<python-input-0>", line 7, in __getattr__
    raise AttributeError("not found")
AttributeError: not found. Did you mean: 'a'?
```

Reply:
Haha, I have the same (i.e., `AttributeError: not found. Did you mean: 'a'?`) in Python 3.11.3... when I run `spin python` but *not* when I run `spin ipython` (I only get `AttributeError: not found`). :sweat_smile: